### DISCUSSION TITLE
are cross-platform lockfiles deprecated?

### DISCUSSION THREAD

**Discussion by DetachHead** [pdm-project/pdm#3037] [Q&A] [answered]
after updating to pdm 1.17.0 and running `pdm update`, i see the following warning:

```
DEPRECATED: `cross_platform` strategy is deprecated in favor of the new lock targets.
See docs: http://pdm-project.org/en/latest/usage/lock-targets/
🔒 Lock successful.  
```

however there doesn't seem to be any mention of this deprecation [in that linked page in the docs](https://pdm-project.org/en/latest/usage/lock-targets/). instead it just states the following:

> By default, PDM will try to make a lock file that works on all platforms within the Python versions specified by [`requires-python` in `pyproject.toml`](https://pdm-project.org/en/latest/usage/project/#specify-requires-python). This is very convenient during development. You can generate a lock file in your development environment and then use this lock file to replicate the same dependency versions in CI/CD or production environments.

this functionality is very convenient, so it would be annoying if i instead had to manually create lockfiles for different platforms. if that is the case however, the docs should probably be updated to mention that this default behavior is deprecated.

it would be a shame if this functionality was removed from pdm :(

**Comment by frostming** [comment, marked as answer]
No, it's the cross-platform strategy that is deprecated. By default, the lock file is still created for cross-platform. Explaining in lock targets, the default lock target is `(python=<project_requires_python>, platform=none, implementation=none)`, to keep the same default behavior as before. And users can narrow down any of the specifiers to make it "less" cross-platform. That is too say, there is no clear boundary between cross-platform and non-cross-platform, it's just a difference in degree. So we do not want to mention cross-platform.

**Reply by DetachHead** [reply]
thanks for the clarification. no idea where the deprecation warning was coming from then because i never changed the config from the default, but i can't seem to reproduce it anymore so i guess i'll mark this as answered

**Reply by frostming** [reply]
The cross-platform strategy is recorded in lock file and PDM silently fixed that for you. In this case we'd better not showing the warning.
